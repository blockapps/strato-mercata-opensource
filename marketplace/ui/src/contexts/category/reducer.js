import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
