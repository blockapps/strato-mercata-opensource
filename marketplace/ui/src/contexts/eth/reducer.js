import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
