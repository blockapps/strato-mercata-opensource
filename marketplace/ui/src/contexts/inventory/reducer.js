import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
