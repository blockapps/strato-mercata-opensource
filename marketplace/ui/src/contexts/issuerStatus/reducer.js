import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
