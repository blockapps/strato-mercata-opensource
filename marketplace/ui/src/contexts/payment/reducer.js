import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
