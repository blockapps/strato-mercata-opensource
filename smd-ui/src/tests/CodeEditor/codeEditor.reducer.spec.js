import reducer from '../../components/CodeEditor/codeEditor.reducer';
import { extAbi, error } from "./codeEditorMock";
import {
  compileCodeFromEditor,
  compileCodeFromEditorFailure,
  compileCodeFromEditorSuccess,
  contractNameChange,
  changeCreateActionState,
  addNewFileTab,
  removeTab,
  onTabChange,
  onChangeFileName,
  onCompileFileLocally
} from '../../components/CodeEditor/codeEditor.actions';
import { deepClone } from '../helper/testHelper';

describe('CodeEditor: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('compile code', () => {

    // CODE_EDITOR_COMPILE_REQUEST
    test('request', () => {
      const action = compileCodeFromEditor('code');
      const initialState = {
        codeCompileSuccess: undefined,
        abi: undefined,
        fileName: undefined,
        createDisabled: true,
        enableCreateAction: false,
        sourceCode: undefined,
        contractName: undefined,
        tab: [{
          text: '',
          title: 'Main.sol'
        }],
        lastTabSelected: 0,
        currentTabSelected: 0,
        isRemoveTab: false,
        localCompileException: ''
      };
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // CODE_EDITOR_COMPILE_SUCCESS
    test('success', () => {
      const action = compileCodeFromEditorSuccess(extAbi);
      const initialState = {
        codeCompileSuccess: undefined,
        abi: undefined,
        fileName: undefined,
        createDisabled: true,
        enableCreateAction: false,
        sourceCode: undefined,
        contractName: undefined,
        tab: [{
          text: '',
          title: 'Main.sol'
        }],
        lastTabSelected: 0,
        currentTabSelected: 0,
        isRemoveTab: false,
        localCompileException: ''
      };
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // CODE_EDITOR_COMPILE_FAILURE
    test('failure', () => {
      const action = compileCodeFromEditorFailure(error);
      const initialState = {
        codeCompileSuccess: undefined,
        abi: undefined,
        fileName: undefined,
        createDisabled: true,
        enableCreateAction: false,
        sourceCode: undefined,
        contractName: undefined,
        tab: [{
          text: '',
          title: 'Main.sol'
        }],
        lastTabSelected: 0,
        currentTabSelected: 0,
        isRemoveTab: false,
        localCompileException: ''
      };
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

  })

  // EDITOR_CONTRACT_NAME_CHANGE
  test('change contract name', () => {
    const action = contractNameChange('Cloner');
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

  // CODE_EDITOR_CHANGE_CREATEACTION
  test('change create action state', () => {
    const action = changeCreateActionState(false, 'abc', 0);
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

  // ADD_NEW_TAB
  test('create new tab', () => {
    const action = addNewFileTab('New File', {
      text: '',
      title: 'Main2.sol'
    });
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

  // REMOVE_TAB
  test('remove tab at index', () => {
    const action = removeTab(0);
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

  describe('on tab change with isRemoveTab value', () => {
    // ON_TAB_CHANGE, isRemoveTab:false
    test('false', () => {
      const action = onTabChange(0, 1);
      const initialState = {
        codeCompileSuccess: undefined,
        abi: undefined,
        fileName: undefined,
        createDisabled: true,
        enableCreateAction: false,
        sourceCode: undefined,
        contractName: undefined,
        tab: [{
          text: '',
          title: 'Main.sol'
        },
        {
          text: '',
          title: 'Main2.sol'
        }],
        lastTabSelected: 0,
        currentTabSelected: 0,
        isRemoveTab: false,
        localCompileException: ''
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // ON_TAB_CHANGE, isRemoveTab: true
    test('true', () => {
      const action = onTabChange(0, 1);
      const initialState = {
        codeCompileSuccess: undefined,
        abi: undefined,
        fileName: undefined,
        createDisabled: true,
        enableCreateAction: false,
        sourceCode: undefined,
        contractName: undefined,
        tab: [{
          text: '',
          title: 'Main.sol'
        },
        {
          text: '',
          title: 'Main2.sol'
        }],
        lastTabSelected: 0,
        currentTabSelected: 0,
        isRemoveTab: true,
        localCompileException: ''
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

  })

  // CHANGE_FILE_NAME
  test('change file name', () => {
    const action = onChangeFileName('New file name');
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

  // ON_COMPILE_FILE_LOCALLY
  test('compile file locally', () => {
    const action = onCompileFileLocally(error);
    const initialState = {
      codeCompileSuccess: undefined,
      abi: undefined,
      fileName: undefined,
      createDisabled: true,
      enableCreateAction: false,
      sourceCode: undefined,
      contractName: undefined,
      tab: [{
        text: '',
        title: 'Main.sol'
      }],
      lastTabSelected: 0,
      currentTabSelected: 0,
      isRemoveTab: false,
      localCompileException: ''
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  })

})
