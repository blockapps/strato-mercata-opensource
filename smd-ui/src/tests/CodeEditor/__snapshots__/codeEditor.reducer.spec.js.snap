// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor: reducer change contract name 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": "Cloner",
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer change create action state 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": "abc",
  "tab": Array [
    Object {
      "text": "abc",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer change file name 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": "New file name",
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer compile code failure 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": false,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "error": "src:6:77: Error: Expected token Semicolon got 'RBrace'
{ Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }
                                                                            ^
",
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "response": " { Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }                                                                             ^",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer compile code request 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "codeType": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "response": "Uploading Contract...",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer compile code success 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": true,
  "contractName": "Cloner",
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": true,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "response": Object {
    "src": Object {
      "Cloner": Object {
        "funcs": Object {
          "cloneSheep": Object {
            "args": Object {
              "name": Object {
                "dynamic": true,
                "index": 0,
                "type": "String",
              },
            },
            "selector": "0072c6a9",
            "vals": Object {
              "#0": Object {
                "index": 0,
                "type": "Address",
              },
            },
          },
        },
        "vars": Object {
          "clones": Object {
            "atBytes": 0,
            "dynamic": true,
            "entry": Object {
              "typedef": "Sheep",
            },
            "public": true,
            "type": "Array",
          },
          "sheepDNA": Object {
            "atBytes": 32,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
        },
      },
      "Sheep": Object {
        "constr": Object {
          "_dna": Object {
            "dynamic": true,
            "index": 1,
            "type": "String",
          },
          "_name": Object {
            "dynamic": true,
            "index": 0,
            "type": "String",
          },
        },
        "funcs": Object {
          "geneticallyModify": Object {
            "args": Object {
              "_dna": Object {
                "dynamic": true,
                "index": 0,
                "type": "String",
              },
            },
            "selector": "0e3668f8",
            "vals": Object {},
          },
        },
        "vars": Object {
          "dna": Object {
            "atBytes": 32,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
          "name": Object {
            "atBytes": 0,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
        },
      },
    },
  },
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer compile file locally 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "src:6:77: Error: Expected token Semicolon got 'RBrace'
{ Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }
                                                                            ^
",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer create new tab 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 1,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": Object {
    "text": "",
    "title": "Main2.sol",
  },
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
    Object {
      "text": Object {
        "text": "",
        "title": "Main2.sol",
      },
      "title": "New File.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer on tab change with isRemoveTab value false 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 1,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": "",
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
    Object {
      "text": "",
      "title": "Main2.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer on tab change with isRemoveTab value true 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": "",
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
    Object {
      "text": "",
      "title": "Main2.sol",
    },
  ],
}
`;

exports[`CodeEditor: reducer remove tab at index 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": true,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": undefined,
  "tab": Array [],
}
`;

exports[`CodeEditor: reducer set initial state 1`] = `
Object {
  "abi": undefined,
  "codeCompileSuccess": undefined,
  "codeType": "SolidVM",
  "contractName": undefined,
  "createDisabled": true,
  "currentTabSelected": 0,
  "enableCreateAction": false,
  "fileName": undefined,
  "isRemoveTab": false,
  "lastTabSelected": 0,
  "localCompileException": "",
  "sourceCode": undefined,
  "tab": Array [
    Object {
      "text": "",
      "title": "Main.sol",
    },
  ],
}
`;
